### PR TITLE
[FLINK-28660][runtime] Simplify logs of blocklist

### PR DESCRIPTION
**Pull request by wanglijie95**
## What is the purpose of the change
Simplify logs of blocklist

## Verifying this change
This change is already covered by existing tests.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (**no**)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (**no**)
  - The serializers: (**no**)
  - The runtime per-record code paths (performance sensitive): (**no**)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn, ZooKeeper: (**no**)
  - The S3 file system connector: (**no**)

## Documentation

  - Does this pull request introduce a new feature? (**no**)
  - If yes, how is the feature documented? (**not applicable**)

### CODE REVIEW

File: flink-runtime/src/main/java/org/apache/flink/runtime/blocklist/DefaultBlocklistHandler.java [hunk 1]
```diff
@@ -111,26 +111,27 @@ private void assertRunningInMainThread() {
...
-                        newlyAddedOrMerged.size(),
+                        "Newly added {} blocked nodes. Total {} blocked nodes currently.",
+                        newlyAddedNodes.size(),
                         blocklistTracker.getAllBlockedNodes().size());
             }
 
-            blocklistListeners.forEach(
-                    listener -> listener.notifyNewBlockedNodes(newlyAddedOrMerged));
-            blocklistContext.blockResources(newlyAddedOrMerged);
+            Collection<BlockedNode> allNodes = result.getAllNodes();
```
Comment (zhuzhurk) [inline flink-runtime/src/main/java/org/apache/flink/runtime/blocklist/DefaultBlocklistHandler.java:132]: maybe we only need to invoke `blockResources` on `newlyAddedNodes`?

Comment (zhuzhurk) [reply flink-runtime/src/main/java/org/apache/flink/runtime/blocklist/DefaultBlocklistHandler.java:132]: Then the `getAllNodes()` is not needed

Comment (wanglijie95) [reply flink-runtime/src/main/java/org/apache/flink/runtime/blocklist/DefaultBlocklistHandler.java:132]: Good idea!

**Review by zhuzhurk — APPROVED** [review]
LGTM.

**Comment by wanglijie95** [comment]
@flinkbot run azure

**Pull request closed without merging**